pkg/utils/ice_config_cache.go

```diff
@@ -0,0 +1,78 @@
+package utils
+
+import (
+	"sync"
+	"time"
+
+	"github.com/livekit/protocol/livekit"
+	"go.uber.org/atomic"
+)
+
+const (
+	iceConfigTTL = 5 * time.Minute
+)
+
+type iceConfigCacheEntry struct {
+	iceConfig  *livekit.ICEConfig
+	modifiedAt time.Time
+}
+
+type IceConfigCache struct {
+	lock    sync.Mutex
+	entries map[interface{}]*iceConfigCacheEntry
```

Comment:
can we use `iceConfigCacheKey` directly instead of interface?

Reply:
using a different key in `cloud`

Reply:
Tried
```
  type IceConfigCache[T any] struct {                                                                                               
      lock    sync.Mutex                                                                                                            
E     entries map[T]*iceConfigCacheEntry // E: invalid map key type T (missing comparable constraint)                               
                                                                                                                                    
      stopped atomic.Bool                                                                                                           
  }
```
but a type of `any` for map key does not work. Is it possible to make it work with a generic type?

Reply:
Thank you @cnderrauber. Did it in this commit - https://github.com/livekit/livekit/pull/2654/commits/5ffbc66a81b4c0ba5e2d0d898155905fab114007